platforms/kotlin/src/main/kotlin/ai/snips/nlu/ontology/ffi/COntology.kt

```diff
@@ -53,20 +54,55 @@ class CIntentParserResult(p: Pointer) : Structure(p), Structure.ByReference {
 
 }
 
-class CIntentClassifierResult : Structure(), Structure.ByReference {
+class CIntentClassifierResult(p: Pointer?) : Structure(p), Structure.ByReference {
+
     @JvmField var intent_name: Pointer? = null
     @JvmField var confidence_score: Float? = null
 
+    init {
+        read()
+    }
+
+    constructor(): this(null)
+
     override fun getFieldOrder() = listOf("intent_name", "confidence_score")
 
     fun toIntentClassifierResult() = IntentClassifierResult(intentName = intent_name?.readString(),
                                                             confidenceScore = confidence_score!!)
 }
 
-class CSlots : Structure(), Structure.ByReference {
+class CIntentClassifierResultList(p: Pointer?) : Structure(p), Structure.ByReference {
```

Comment:
name it `CIntentClassifierResults` to follow the exisiting convention ? (we use `s` in nlu ontology and `Array` in `hermes`... let's not introduce a new one)